src/mon/OSDMonitor.cc

```diff
@@ -4837,6 +4839,34 @@ void OSDMonitor::check_pg_creates_sub(Subscription *sub)
   }
 }
 
+void OSDMonitor::check_for_filestore_osds(health_check_map_t *checks)
```

Comment:
There is already a helper `is_pool_currently_all_bluestore()`, can we just use that?

Reply:
Thanks Neha. I checked the `OSDMonitor::is_pool_currently_all_bluestore`, it checks OSDs associated with only few PGs (8) for specific pool if they are on bluestore. We need to modify `is_pool_currently_all_bluestore` to check for all OSDs and get OSD list which are not on bluestore to iterate over all pools. We want to list down all OSDs which are on filestore in health detail. Let me know your thoughts.

Reply:
my bad, I was thinking of `has_filestore_osd`, which isn't a function of its own yet.